axlearn/common/config.py

```diff
@@ -387,6 +387,8 @@ def process_kv(key: str, val: Any):
                 default_val = field.default
                 if val is default_val and default_val in omit_default_values:
                     return
+            elif field is None and val in omit_default_values:
```

Comment:
I wonder if we could apply this logic only to dataclass fields.

Reply:
changed